### PR TITLE
Toxins tweaks - Box Station

### PR DESCRIPTION
**Pull request by eplgr**
_PSA: if you see any non functional work-environment in your department and you want it changed, you can post here or ping me at discord. I'll analyze it and try to improve it if it's needed._

# Github changelog
**Box station:**
- Replaced a table with a pipe dispenser
- Moved the screwdriver and wrench from the replaced table to a previously empty table
- Changed Toxins Storage to properly function like a storage: added an unwrenched freezer, a heater, a mechanical toolbox and a hazard vest. Also added a firefighter locker.
- Damaged floor for flavour.

# Reasons & personal thoughts
This is long overdue. You can't really do any good job at toxins without ~a RPD~ _building pipes_, ~and normally you'd ask an engineer/atmos tech to get one for you, if you find one. I do not consider that kind of department interaction good or vital to the game mainly because we are not talking about cargo, the kitchen, bar, or any other department that really have that idea behind them, we are talking about engineering.~

And finally, I think all other maps have these tools at toxins, so why leave behind Box? Also repurposing Toxin's Storage Room seems nice, it felt like a waste of space before.

# In-game changelog
:cl: 
 * tweak: Tweaks to Box's Toxins and its storage: added a few tools and machines to improve its functionality

### CODE REVIEW

**Comment by eplgr** [comment]
Top honk, I fucked up. Will report back shortly.

**Comment by FudgePucker** [comment]
👍 Normally I would go on a ten paragraph essay about how science shouldn't receive any new shit because how feature bloated it is.
But Toxins I feel like is one of the most neglected parts of science besides xenoarch and xenobio.
With how it is right now if you want to learn how to make bombs you would just go atmotech and just hack the one airlock needed to get the TTVs, then you would have access to the main air tanks, a heater and freezer, a hardsuit that can survive the surface of the sun, and as many RPDs as you need.

Besides, if they really wanted to they would be able to make a RPD and heater, adding it in already doesn't make it easier to bomb the shit out of the station, it makes it easier on people experimenting, when you get DAT MOTHA FUCKIN TATOR ROLL OH BABY STEP ASIDE AND WATCH HOW A MASTER MAKES SIX MAX DEV BOMBS. When you know how to make bombs it doesn't take that much time, when you are experimenting you need all the time you can get from a round.

So in summary, it won't make the job of 'blowing up the station following simple instructions' that much easier, but makes the chore of figuring out how bombs really work much less tedious. You can't just kill an engineer for his ID as non-antag.

**Comment by SonixApache** [comment]
Replace RPD with a normal pipe dispenser.

**Comment by Sprok0** [comment]
Scientists can already print an RPD after hacking their roundstart autolathe, why give them one for free

**Comment by eplgr** [comment]
@SonixApache No. Pipe dispensers are outdated. The other day there even was a chat where everyone agreed there is no reason for them still being in the game, but no one really bothers to actually remove them.

@Sprok0 **Functionality purposes**. It's not a good idea to restrict items that are mandatory for jobs. Now, on top of that, you are suggesting these items should be in hacked machines; let me remind you that RPDs are extremely expensive to print (and metal&glass are not actually found in excessive amounts in science) and that there's a big chance you'll get shocked.

**Comment by Riekland** [comment]
I would replace one of the tiles in there with an exploded one for flavour tbdesu

**Comment by SonixApache** [comment]
@eplgr Pipe dispensers aren't outdated; you're just so extremely accustomed to the (supposedly rare) RPD babby mode that you can't think of any other way. And it's a decent downside to getting easy access to easy near-guaranteed shuttlecalling tools.

Getting the RPD from the lathe might take some discussing with the scientist staffing R&D (aka nobody) or might require you to acquire some extra minerals. I don't think a minute of feeding and hacking the lathe is that big a deal for the pay-off you get. Remember that the station is supposed to require some setup.

Also, you have to be pretty bad to get shocked while hacking a lathe.

**Comment by eplgr** [comment]
@SonixApache Pipe dispensers have not been used in ages, they are just very, very non-functional. I don't see why you think adding them to ONLY Box Station's Toxins is a good idea. 
If anything, I suggest you make a new issue thread stating that all other maps should remove their RPD's from toxins.
You could even try to find out if these maps have a higher % of bombs destroying the station just because they start with RPDs! 

>Remember that the station is supposed to require some setup.

These kind of comments are what I was trying to avoid when I talked about departament interactions. Yes, you are right, but what you are proposing is just bad design. The "required set up" is already here: people actually having to build their own pipes set up.

**Comment by SonixApache** [comment]
If the other maps already have RPDs, I'm cool with it.
I'll make my own PR when I get sick of blowing up.

**Comment by FudgePucker** [comment]
I agree with SonixApache.
If I remember correctly the pipe dispenser works the exact same as the RPD, only it is a giant machine instead of a handy tool that fits in your backpack.

RPD is more useful for engineers and sabotage because of its portability, you can make an entire station worth of pipes from a single tool that fits in your bag. You aren't going to be repairing bomb damage as a toxins researcher, you cause bomb damage, so having a stationary 'rpd' is good enough.

But its not that big of a deal, if you wanted to fuck up atmospherics on the station you have several canisters of air and can make TTV bombs.

EDIT: Also it being stationary means that fuckface powergamer in research isn't going to pocket it. People are more likely to loot a RPD instead of that giant machine.

**Comment by eplgr** [comment]
I just don't think that having to drag pipes around and having to rotate them one by one and so on adds richness to the game.

**Comment by SonixApache** [comment]
I don't think having my department and body turned into nothingness adds to the richness to the game, but it's a risk I'm willing to take.

**Comment by FudgePucker** [comment]
@eplgr 
I still think my argument of 'that one asshole powergamer who picked science to valid hunt can't just stash it in his backpack "Just in case"' still stands though.

On one hand it is somewhat of a pain to pick up then rotate the pipe, on the other hand you will still have access to it if you didn't rush to toxins at round start to secure it before any dipshits did. The giant machine forces people to share it. Good luck convincing that one asshole in research to let you borrow it to put down some pipes, he might retaliate by spacing your body because you were probably valid.

**Comment by FudgePucker** [comment]
And trust me, 'that one asshole powergamer' plays every round, I would say I described 50% of all science players with that sentience.

**Comment by eplgr** [comment]
I think I fixed the issues and there are no more conflicts.

Also, changed a few things.

**Comment by BobdaBiscuit** [comment]
What's the reasoning behind changing the plasma canister for an oxygen one?

Otherwise really solid changes.

**Review by Probe1 — CHANGES_REQUESTED** [review]
No RPDs on the dance floor.  If you want me to write my own long balance discussion over it I can but everyones pretty much made it clear why.  It's a video game just because you can get something doesn't mean you should get it at round start.

**Comment by Shadowmech88** [comment]
>just because you can get something

What if that thing is so necessary to be effective at a given job that players feel the need to go out of their way to acquire it nearly every round, to the point where it just becomes a chore that one must do before starting on their real job instead of simply an option that one can choose to do?

**Comment by Probe1** [comment]
You don't need an atmos only item to do toxins.  You want one because you want to make a perfect bomb. 

Sorry but no.  Mapping is about giving the players what they need so they can build on that.  This is absolutely naked powercreep.

**Comment by BobdaBiscuit** [comment]
>You don't need an atmos only item to do toxins.

Yes you do actually, because toxins is an atmos based job.

**Comment by Probe1** [comment]
Which is why you're given everything you need and an autolathe to get what you want.  

Needs versus wants.  Make the necessary adjustments.  A pipe dispenser is an acceptable downgrade.

**Comment by eplgr** [comment]
@Probe1 No, you want one because you want to _learn_. 
I believe the following: rpds don't directly influence players into making more bombs. The other maps already have them and it's not like we are having massive bombings very often there. Whether a player decides to actually bomb the station or not has nothing to do with RPDs being in toxins at roundstart.

Also, I really find it odd that some people are so against this, but actually never openly said anything about RPDs and other stuff already appearing in other map's toxins.
In fact, the only issue/pr created regarding RPDs is this: https://github.com/d3athrow/vgstation13/issues/6167 , apparently it got closed because he didn't do anything about the change requests, and his design was probably not perfect, but he said the following regarding this topic:
>To the people that will complain that oh no everything is megabombs now! It already is. You dip your head in you know where, flip a pump, collect your cool gas. Flip the other, collect your hot gas. Go jam it into a valve and bombcap ahoy.
**A better lab makes legitimate plasma research far less painful when you have more time to tweak and test rather than setting up whatever pieces of a better lab you need to work on whatever you wanted to research.**

Now, I do not agree that toxins is out of control already, but for sure the aim of this PR is to provide better tools for functionality.

I'd even argue that we're seeing less and less ttv bombings. If you want to do stuff like that, you just steal the shard and let it delaminate, which is what I've seen A LOT recently. I don't remember the last time I saw a ttv bomb exploding in the station.

**Comment by Probe1** [comment]
Other maps get other things because they're non standard.  Even Dragonbro that added them in has said it was a mistake and he was strong armed into adding it by @SarahJohnson489 who is by no means a "learning" player.

You certainly don't see us adding in thermals on all maps because Metaclub has them.

You're not making a convincing case by saying no one ever blows up the station when that happens almost daily.  Or that an RPD is essential, or that spending 15 seconds acquiring an RPD is a burden too great for advanced players who want to get max cap bombs to bear.

**Comment by BobdaBiscuit** [comment]
>Which is why you're given everything you need and an autolathe to get what you want.
>you're given everything you need

That's just a lie.

Box by far has the worst toxins lab from all 3 maps, and has almost no room at all for expansion so all pipework has to be done in space, at the outpost, or cluttered around science.
Learning the subject properly needs a lot of wiggle room to try out different mixes and locking it behind an autolathe which needs gloves/an AI to hack just makes it so people who might have gotten into the subject just drop it cause they can't be assed.
There's also the point that people might not know that RPDs can be obtained from hacked lathes since it's a bit obscure.

**Comment by BobdaBiscuit** [comment]
Let's just remove the round stand shard and singulo devices and make engis order them from cargo tbh, you don't need them to power the station. They just want them to make the perfect engine.
[I'm meming at this point but the point stands]

**Comment by eplgr** [comment]
I consider this a QoL change, we won't see more bombs causing mayhem because of it and if people decide to go learn more toxin as a result, I'd say that's good. Other than this, I am not really sure what so say.
Is there any chance we actually _test_ how this turns out? We can always delete the rpd/revert it.

@BobdaBiscuit If there's gonna be a canister in there, I think it's more useful if it's an O2 one because you can just straight connect it to whatever pipe set up you are making at round start, start cooling it, or whatever you want and so on. The plasma canister was either ignored or just saved you 1 of the 3-4 trips you have to make to the gas storage. 
I am open to suggestions, of course.

**Comment by BobdaBiscuit** [comment]
It's just personal preference, I use it often for topping up tanks with since it's near the burn chamber output, also nice having an extra plasma canister since they are rarish around the station.

**Comment by FudgePucker** [comment]
@Probe1 
There is some flaws with your reasoning.
1) You can still get max cap bombs with out using an RPD
2) Only research is really a giant power creep monstrosity that should be taken out back and shot, the rest of science is pretty okayish. Adding a way to make pipes without having to deal with hacking a machine in research is not going to destroy the meta. Also I wouldn't want to hack the autolathe in research anyways, if you accidentally set it to shock, that one powergaming asshole in research (50% of all science players) will space you because you are probably valid in his mind.

**Comment by Exxion** [comment]
Fudge, are you sure you read Probe's posts correctly?

**Comment by eplgr** [comment]
I am gonna add the pipe dispenser instead, which of course will require some other changes, there's not much free space around. After giving it a try it's not so bad and I actually like that it draws more attention.
We'll see.

Thanks for the feedback.

**Comment by eplgr** [comment]
What do you people think? Also, updated the OP stating what exacly changed.

**Comment by FudgePucker** [comment]
@Exxion I am pretty sure I did.
You need an RPD to experiment and try to figure out fancy things like making bombs with N2 and N2O.
You don't need an RPD to make six big enough to ruin the station bombs 15 mins into the round.

Although the Pipe dispenser would be a good compromise. Mostly because it does what is needed and can't be pocketed by someone else for sabotage.

@eplgr Lookin good.

**Review by Probe1 — APPROVED** [review]

**Comment by PJB3005** [comment]
Oh I just realized something.

The pipe dispenser can't do layers. May want to consider a pipe alignment frame.

**Comment by eplgr** [comment]
@PJB3005 I did test the map changes and I saw the layer option, but you made me doubt whether it was working or not.
So now I tested it again: it's there and it does work.



At the bottom, you can find the option:

**Comment by PJB3005** [comment]
Wait really?

Huh well that shows how much I've used the pipe dispenser since layers got added.

(Exactly once to set up an artifact gas chamber when all the RPDs in engineering were gone)

**Comment by ghost** [comment]
@SonixApache 

Just to clear the air:
* The advantages of an RPD over a pipe dispenser is being able to place pipes on walls without deconstructing the, often reinforced, wall and a ui that doesn't make you want to kill yourself when using layers. Heavy repiping jobs like , https://cloud.githubusercontent.com/assets/5564796/10533835/46ad39ac-737b-11e5-8b6d-6d00a1702ba1.png , or any major atmos work involving moving whole lockers of pipes around, fishing them out with alt-click, and dragging in sequence/manually rotating them is terrible and only people who haven't time and time again would think it's alright.
* You can get around this in the toxin's lab by simply opening the burn chamber and putting the thermals in the central slot or putting valved heat exchangers or filters on the output of the burn chamber which is most of what you'd add anyway. The chiller is trivial and you can either chair or mag some anyway. You don't need much gas out of it so it's not a big deal.
* The key to a bomb's strength is ratio of how much of what to what in what. You can make 100C plasma, or whatever your pepperidge farm cookie mix wants, in the bathroom with welderfuel, some stolen pipes/pumps, and scrubbed plasma tanks, hack 1 door while pulling a backpack of tanks then go blow up the station with max caps all without either an rpd or a dispenser. You don't need super fancy mega-hot plasma or even pure plasma to hit cap once you work out your probable result mix ranges to still hit cap. There are 3 existing burn chambers and nobody ever had issues with getting chilled gas even if they had to work out of cryo freezer oxygen cookie mixes. Having a better facility helps people learn, not make max caps because they already have everything they need to make fast maxcaps. Good luck getting an uncap for giggle 'n' griff, blob, longer reach on anti-xenos/Queen spider blast cannons, or even wizlingpire denial. bhangmeter autism is otherwise irrelevant to the average round. The challenge is always in developing a recipe, once you have it you're lightning fast and are fine with far cruddier facilities as you start increasing efficiency.
* A lack of an RPD doesn't stop anyone from using an IED on exposed pipes to put some lifted public waste manifolds and connectors into distro. It doesn't stop people from being able to take a short trip for a meme engine.
* Do NOT remove the atmos pipe dispenser, it's key for standard cyborgs to be able to setup power and kind of fix shit.

A pipe dispenser saves the dept from people wasting like 30 metal and 15 glass for light reworks, it and the heater/freezer saves the researcher a trip so they can get right into testing faster, and it's trivial for people that care to do more to do the kelo-derma we've done for years. It's a win-win.

tl;dr, Knowledge is the height of the boom.ogg bar, not the facility. You can already quickly make maxcap co2 bombs all over faster than sec can get their jammies on, a pipe dispenser isn't the harbinger of the boom.oggcalypse unless the boom.oggcalypse is a very salty autist bitching over common because the round ended in under 50 minutes 7 times in a row before they could do/complete their tests.
